### PR TITLE
Pause button for autoplay #156

### LINKED ISSUE #156: Pause button for autoplay videos
Blocks with an autoplay video should have a pause button, to comply with accessibility standards. Specifically, we’re mitigating [Failure 93 of Success Criterion 1.4.2](https://www.w3.org/TR/WCAG20-TECHS/F93.html).

Designs have been adjusted to show a pause button. For now, we’re focusing on the hero #41, [the design is available](https://www.figma.com/file/yv0VtHTvMoGPqkv0RAYcVp/PDP?type=design&node-id=6568-48961&mode=design&t=Rr7XpNdhodaMcgfL-4). 

- The video player for autoplay videos should have a visible pause button when the video starts playing
- The button must be discernible from other controls and be clearly labeled and icon that indicates its functionality
- The pause/play button should be accessible to users navigating via keyboard, meaning it should be focusable and actionable using keyboard inputs. The focus style should be the same as for the scroll to top button
- Upon pressing the pause button, the autoplay video should immediately stop playing
- Once the video is paused, the [pause button should change to a play button](https://www.figma.com/file/Id3nd9kbfh8TpS1SfWhrW2/Volvo-Component-Library?type=design&node-id=566-2770&mode=dev)
- The button should be large enough to be easily tapped on touch devices without accidentally pressing adjacent controls

### PR DESCRIPTION
**Pull request by markovukiceviccn**
Fix #156 

Test URLs:
- Before: https://develop--vg-volvotrucks-us-rd--netcentric.hlx.page/
- After: https://156-pause-button-for-autoplay--vg-volvotrucks-us-rd--netcentric.hlx.page/

### CODE REVIEW

File: blocks/v2-hero/v2-hero.js [hunk 1]
```diff
@@ -35,8 +27,56 @@ export default async function decorate(block) {
...
+    `);
+
+    playPauseButton.appendChild(pauseIconFragment);
+    playPauseButton.appendChild(playIconFragment);
+    contentWrapper.appendChild(playPauseButton);
+
+    const playIcon = block.querySelector('.icon-play-video');
+    const pauseIcon = block.querySelector('.icon-pause-video');
+
+    playPauseButton.setAttribute('aria-label', 'Pause video');
```
Comment (TomaszDziezykNetcentric) [inline blocks/v2-hero/v2-hero.js:64]: Could you use `getTextLabel`?

File: blocks/v2-hero/v2-hero.js [hunk 2]
```diff
@@ -35,8 +27,56 @@ export default async function decorate(block) {
...
+
+    // Toggle the play/pause icon on click
+    playPauseButton.addEventListener('click', () => {
+      const isPaused = video.paused;
+      video[isPaused ? 'play' : 'pause']();
+      if (!isPaused) {
+        pauseIcon.style.display = 'none';
+        playIcon.style.display = 'flex';
+        playPauseButton.setAttribute('aria-label', 'Play video');
```
Comment (TomaszDziezykNetcentric) [inline blocks/v2-hero/v2-hero.js:73]: Could you use getTextLabel?

File: blocks/v2-hero/v2-hero.js [hunk 3]
```diff
@@ -35,8 +27,56 @@ export default async function decorate(block) {
...
+      } else {
+        pauseIcon.style.display = 'flex';
+        playIcon.style.display = 'none';
+        playPauseButton.setAttribute('aria-label', 'Pause video');
```
Comment (TomaszDziezykNetcentric) [inline blocks/v2-hero/v2-hero.js:77]: Could you use getTextLabel?

**Review by TomaszDziezykNetcentric — CHANGES_REQUESTED** [review]

File: blocks/v2-hero/v2-hero.js [hunk 4]
```diff
@@ -35,8 +27,56 @@ export default async function decorate(block) {
     block.prepend(video);
```
[the remark this answers is not in the document]
Comment (markovukiceviccn) [reply blocks/v2-hero/v2-hero.js:29]: Playsinline and loop attributes has been added.

File: blocks/v2-hero/v2-hero.js (hunk 3 above)
Comment (markovukiceviccn) [reply blocks/v2-hero/v2-hero.js]: getTextLabel has been utilised.

File: blocks/v2-hero/v2-hero.css [hunk 5]
```diff
@@ -131,6 +131,37 @@
...
+  display: flex;
+  justify-content: center;
+}
+
+.v2-hero .v2-hero__playback-button {
+  position: absolute;
+  bottom: 20px;
+  right: 20px;
+  background: transparent;
+  width: 32px;
```
[the remark this answers is not in the document]
Comment (markovukiceviccn) [reply blocks/v2-hero/v2-hero.css]: Support for the "fat fingers" has been added.

**Review by TomaszDziezykNetcentric — APPROVED** [review]

**Pull request merged**